### DISCUSSION TITLE
update css element of a region based on id

### DISCUSSION THREAD

**Discussion by rioharper** [katspaugh/wavesurfer.js#3121] [Q&A]
Hello (again!) I have a quick question. Below is a class that basically creates a whole bunch of regions from text file info. What I would like is to when the user passes over an already-created region, for it to change the border radius of that region (i.e active) and then revert back to it's normal css settings when the cursor leaves the region (i.e inactive).

Currently, the code has no problem customizing the css for creating regions (addRegion) and changing id's of regions to active/inactive, but it makes no change to the css styling on the waveform. Is there something I'm missing? Thanks!

```javascript
/*
  <html>
    <style>
      #waveform ::part(active) {
        border-radius: 20px !important;
        height: 50% !important;
      }

      #waveform ::part(inactive) {
        border-radius: 10px !important;
      }
*/
class RegionsManager {
  median;
  stdDev;
  activeRegion;
  constructor(regions, text) {
    this.regions = regions;
    this.text = text;
    this.lines = [];
    this.createRegions();
    this.setupInteractions();
  }

  setupInteractions() {
    this.regions.on("region-clicked", (region, e) => {
      if (e.shiftKey) {
        console.log('Shift key pressed');
        e.stopPropagation();
        this.activeRegion = region;
        region.play();
        } else {
          this.activeRegion = null;
          }
        });
    this.regions.on("region-in", function(region) {
      console.log('Region in');
      region.id = "active";
      }.bind(this))

    this.regions.on("region-out", function(region) {
      region.id = "inactive";
    }.bind(this))
  }
  

  addRegionsFromLines(lines) {
    lines.forEach(line => {
      try {
        const [start, end, confidence, content] = this.extractLineData(line);
        this.regions.addRegion({
          id: "inactive",
          start: parseFloat(start),
          end: parseFloat(end),
          color: this.getConfidenceColor(confidence),
        });
      } catch (error) {
        console.error(`Error adding region for line: ${line}`, error);
      }
      console.log('Regions added successfully');
    });
  }
}```

**Comment by katspaugh** [comment]
Your CSS is correct but you cannot change the id of a region like that. It just sets the property of the object but doesn't change it in the DOM.

Please feel free to fix this in the Regions code by adjusting the `setOptions` method. If an id is passed, it should add it to the part attribute.

**Comment by rioharper** [comment]
#3124 this should add that! (and a few more features)
